core/common/src/main/java/alluxio/underfs/UnderFileSystem.java

```diff
@@ -37,127 +37,44 @@
  */
 @ThreadSafe
 // TODO(adit); API calls should use a URI instead of a String wherever appropriate
-public interface UnderFileSystem {
+public interface UnderFileSystem extends Closeable {
 
   /**
    * The factory for the {@link UnderFileSystem}.
    */
   class Factory {
-    private static final Cache UFS_CACHE = new Cache();
```

Comment:
as talked offline with @gpang  and @peisun1115 , we try to simplify the factory method of UnderFileSystem , and move the intelligence like caching/indexing to `UfsManager`